doc/_scripts/software_maturity_scanner.py

```diff
@@ -0,0 +1,554 @@
+use the ``--commit-sha`` option upon execution to mark the commit on which the
+build output is based.
+
+Script Logic
+************
+
+A complete list of all available SoCs are found by searching through the
+samples and applications folder for sample.yaml files and retrieving the
+``integration_platforms`` list from each. Anything nRF51 and not starting with
+nRF is excluded.
+
+To mark features as experimental the twister output folder is searched for
+all build.log files and scanned for the string "Experimental symbol XXX is
+enabled". When found, The SoC is looked up in /zephyr/.config, and a mapping
+from each SoC to the experimental symbols detected when using it to build is
+stored and used to evaluate the Kconfig rules.
+
+Next, the twister output folder is searched for all .config-files. These
+are evaluated against the given Kconfig rules to determine which features
+are supported, experimental or not supported. If a rule contains multiple
+symbols connected with either && or ||, all of the symbols must be present in
+the same .config-file for a SoC to be registered as supporting that feature.
+That means that a feature is not marked as supported if there is no explicit
+test case where all the required Kconfig symbols are present at the same time,
+even if the required options have been enabled individually in separate
+samples. If at least one of the symbols are marked as experimental, the feature
+is also marked as experimental.
+
+In cases where there are rules with multiple symbols connected with ||, and one
+is experimental and the others are not, the feature will still be marked as
+experimental. If this is an undesired outcome, the feature should be split
+into two separate features, one with the experimental symbols and the other
+with the supported ones.
+
+Copyright (c) 2022 Nordic Semiconductor ASA
+"""
+
+from pathlib import Path
+from typing import Dict, List, Set
+from azure.storage.blob import ContainerClient
+from azure.core.exceptions import ClientAuthenticationError
+import kconfiglib
+import os
+import re
+import sys
+import yaml
+import logging
+import argparse
+
+
+AZ_CONN_STR = ";".join(
+    (
+        "DefaultEndpointsProtocol=https",
+        "EndpointSuffix=core.windows.net",
+        "AccountName=ncsdocsa",
+        "AccountKey={}",
+    )
+)
+"""Azure connection string (private acces)"""
+
+AZ_CONTAINER = "ncs-doc-generated-reports"
+"""Azure container."""
+
+UPLOAD_PREFIX = "software_maturity"
+"""Prefix used by all files related to this script."""
+
+UPLOAD_FILENAME = UPLOAD_PREFIX + "/{}.yaml"
+"""Filename of uploaded software maturity level report."""
+
+NRF_SAMPLE_DIR = Path(__file__).absolute().parents[2]
+"""Directory containing samples/ and applications/ subfolders."""
+
+__version__ = "0.1.0"
+
+logger = logging.getLogger(__name__)
+
+
+def upload_data(filename: str, data: dict, cc: ContainerClient) -> None:
+    """Upload software maturity level report to Azure blob storage.
+
+    Args:
```

Comment:
in some docstrings args are present and in some not.

Reply:
All docstrings should now be consistent